### PR TITLE
ARTEMIS-2480 - Reloading configuration can kill broker

### PR DESCRIPTION
**Pull request by andytaylor**
https://issues.apache.org/jira/browse/ARTEMIS-2480

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java [hunk 1]
```diff
@@ -2025,4 +2025,12 @@ void slowConsumerDetected(String sessionID,
    @LogMessage(level = Logger.Level.WARN)
    @Message(id = 224101, value = "Apache ActiveMQ Artemis is using a scheduled pool without remove on cancel policy, so a cancelled task could be not automatically removed from the work queue, it may also cause unbounded retention of cancelled tasks.", format = Message.Format.MESSAGE_FORMAT)
    void scheduledPoolWithNoRemoveOnCancelPolicy();
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(id = 224102, value = "unable to undeply address {0} : reason {1}", format = Message.Format.MESSAGE_FORMAT)
+   void unableToUndeployAddress(SimpleString addressName, String reason);
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(id = 224103, value = "unable to undeply address {0} : reason {1}", format = Message.Format.MESSAGE_FORMAT)
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java]: undepl**o**y

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java [hunk 2]
```diff
@@ -2025,4 +2025,12 @@ void slowConsumerDetected(String sessionID,
    @LogMessage(level = Logger.Level.WARN)
    @Message(id = 224101, value = "Apache ActiveMQ Artemis is using a scheduled pool without remove on cancel policy, so a cancelled task could be not automatically removed from the work queue, it may also cause unbounded retention of cancelled tasks.", format = Message.Format.MESSAGE_FORMAT)
    void scheduledPoolWithNoRemoveOnCancelPolicy();
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(id = 224102, value = "unable to undeply address {0} : reason {1}", format = Message.Format.MESSAGE_FORMAT)
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java]: undepl**o**y

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java [hunk 3]
```diff
@@ -3022,12 +3022,20 @@ private void undeployAddressesAndQueueNotInConfiguration(Configuration configura
...
+            } catch (Exception e) {
+               ActiveMQServerLogger.LOGGER.unableToUndeployAddress(addressName, e.getMessage());
+            }
          } else if (addressSettings.getConfigDeleteQueues() == DeletionPolicy.FORCE) {
             for (Queue queue : listConfiguredQueues(addressName)) {
                if (!queuesInConfig.contains(queue.getName().toString())) {
                   ActiveMQServerLogger.LOGGER.undeployQueue(queue.getName());
-                  queue.deleteQueue(true);
+                  try {
+                     queue.deleteQueue(true);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java]: this should logic should then also exist in line 3022 also.

**Comment by michaelandrepearce** [comment]
test case?

**Comment by andytaylor** [comment]
feedback applied

**Pull request closed without merging**